File: Leetcode/905/Selection_Sort_Algo/main.dart
class Solution {
  List<int> sortArrayByParity(List<int> nums) {
    int n = nums.length;
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            if (nums[j] % 2 == 0) {
                // swap
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                break;
            }
        }
    } 
    return nums;
  }
}

File: Leetcode/905/Selection_Sort_Algo/main.go
func sortArrayByParity(nums []int) []int {
    n := len(nums)

    for i:= 0; i < n; i++ {
        for j:= i+1; j < n; j++ {
            if nums[j] % 2 == 0 {
                // swap
                temp := nums[i]
                nums[i] = nums[j]
                nums[j] = temp
                break
            }
        }
    }

    return nums
    
}